compiler/rustc_const_eval/src/errors.rs

```diff
@@ -437,29 +438,22 @@ pub struct UndefinedBehavior {
     pub raw_bytes: RawBytesNote,
 }
 
-pub trait ReportErrorExt {
-    /// Returns the diagnostic message for this error.
-    fn diagnostic_message(&self) -> DiagnosticMessage;
-    fn add_args<G: EmissionGuarantee>(
-        self,
-        handler: &Handler,
-        builder: &mut DiagnosticBuilder<'_, G>,
-    );
-
-    fn debug(self) -> String
-    where
-        Self: Sized,
-    {
-        ty::tls::with(move |tcx| {
-            let mut builder = tcx.sess.struct_allow(DiagnosticMessage::Str(String::new().into()));
-            let handler = &tcx.sess.parse_sess.span_diagnostic;
-            let message = self.diagnostic_message();
-            self.add_args(handler, &mut builder);
-            let s = handler.eagerly_translate_to_string(message, builder.args());
-            builder.cancel();
-            s
-        })
-    }
+/// args: eagerly_translate_with_args!(
+///     handler: &Handler,
+///     msg: DiagnosticMessage,
+///     key: str,
+///     value: impl IntoDiagnosticArg,
+/// )
+macro_rules! eagerly_translate_with_args {
+    ($handler:expr,$msg:expr$(,$key:expr,$value:expr)*$(,)?) => {
+        $handler.eagerly_translate_to_string(
+            $msg,
+            [
+                $(($key.into(), $value.into_diagnostic_arg()),)*
+            ].iter()
+                .map(|(a, b)| (a, b))
```

Comment:
What does this do? Seems to be an identity map which doesn't do anything?
Please either remove or add a comment explaining its purpose.